src/lib/OpenEXRCore/internal_constants.h

```diff
@@ -28,4 +28,10 @@
 #define EXR_SHORTNAME_MAXLEN 31
 #define EXR_LONGNAME_MAXLEN 255
 
+#if defined(__MSVCRT__)
+#define EXR_PRIu64 "I64u"
```

Comment:
I'm really surprised MinGW's inttypes.h doesn't define PRIu64 even when using non-C99 MSVCRT?!

On my native MSYS2 install (so maybe more recent MinGW), I have in inttypes.h:
```
/* 7.8.1 Macros for format specifiers
 * 
 * MS runtime does not yet understand C9x standard "ll"
 * length specifier. It appears to treat "ll" as "l".
 * The non-standard I64 length specifier causes warning in GCC,
 * but understood by MS runtime functions.
 * crtdll.dll and msvcrt10.dll do not support any 64-bit modifier.
 */
#if defined(_UCRT) || __USE_MINGW_ANSI_STDIO
#define PRId64 "lld"
#define PRIi64 "lli"
#define PRIo64 "llo"
#define PRIu64 "llu"
#define PRIx64 "llx"
#define PRIX64 "llX"
#elif !defined(__CRTDLL__) && __MSVCRT_VERSION__ >= 0x200
#define PRId64 "I64d"
#define PRIi64 "I64i"
#define PRIo64 "I64o"
#define PRIu64 "I64u"
#define PRIx64 "I64x"
#define PRIX64 "I64X"
#endif
```

Reply:
I'd leave PRIu64 stuff as is and live w/ the warning until Ubuntu's cross mingw is updated ;)

Reply:
It does define `PRIu64`, but it defines it as `%llu`, which the compiler then warns isn't recognized. If this really is just an outdated Ubuntu mingw thing, I'm fine just leaving the warning in place.  Just came across it in the course of investigating some other cross-compiler warnings.

Reply:
> If this really is just an outdated Ubuntu mingw thing, I'm fine just leaving the warning in place.

You could try using Fedora as well just to check, as it ships a more recent MinGW toolchain.

Perhaps `__USE_MINGW_ANSI_STDIO` is defined somewhere and is then not falling through to the other case?

I can also check native MinGW building under MSYS2 when I get a chance. In general, it should be handled correctly by inttypes.h and no further intervention should be needed....

Reply:
Alternatively, you can just #undef and redefine PRIu64 if absolutely needed in the MSVCRT case only without touching other code, with a comment that the workaround can/should be removed at some point...

Reply:
I reverted the attempt to fix the `PRIu64` warning altogether, it's not worth the hassle. I'm happy to revisit if someone reports an actual problem with the formatting, but it seems better to live with the a rare warning than a messy attempt to work around it.

Reply:
Makes sense.

Maybe revert the empty new line as well?